packages/push/src/push_vm/stack.rs

```diff
@@ -312,6 +314,51 @@ impl<T> Stack<T> {
         Ok((x, y))
     }
 
+    /// Returns a triple of references to the top three elements of the stack,
+    /// or an error if the stack has less than three elements. The first element
+    /// of the triple will be the top element of the stack, the second element
+    /// of the triple will be the second-to-top element of the stack, and the
+    /// third element of the triple will be the third-to-top element of the
+    /// stack.
+    ///
+    /// # Errors
+    ///
+    /// Returns `StackError::Underflow` if the stack has less than three
+    /// elements.
+    pub fn top3(&self) -> Result<(&T, &T, &T), StackError> {
```

Comment:
I'd approve this for this pr but we should really figure out a way to make this more general like my proposed `top_n` with a potential rework.

Reply:
This is a good idea, but I'm going to push it off to a new issue (#358).